fairlearn/preprocessing/_correlation_remover.py

```diff
@@ -113,11 +116,26 @@ def transform(self, X):
         X_filtered = np.atleast_2d(X_filtered)
         return self.alpha * X_filtered + (1 - self.alpha) * X_use
 
+    def _check_sensitive_features_in_X(self, X) -> None:
+        """Check if the sensitive features are in X."""
+        if isinstance(X, pd.DataFrame):
+            missing_columns = [c for c in self.sensitive_feature_ids if c not in X.columns]
+        else:
+            if X.ndim == 1:
```

Comment:
I'm surprised this doesn't fail since at this point we cannot be sure that X is an array since it's not processed.

Reply:
Indeed I only handled pandas and numpy objects, but that was mainly because I saw it would [break earlier](https://github.com/fairlearn/fairlearn/blob/main/fairlearn/preprocessing/_correlation_remover.py#L82) if the input has another type